frontend/simple/main.js

```diff
@@ -17,7 +18,39 @@ Vue.use(Router)
 Vue.use(VeeValidate)
 
 Vue.events = new Vue() // global event bus, use: https://vuejs.org/v2/api/#Instance-Methods-Events
-
+/*
+  The following are reusable guard for routes
+  the 'guard' defines how the route is blocked and the redirect determines the redirect behavior
+  when a route is blocked
+ */
+// Check if user is logged in
+var loginGuard = {
+  guard (store) {
+    return !store.state.loggedIn
+  },
+  redirect (to, from) {
+    return { path: '/signup', query: { next: to.path } }
+  }
+}
+// Check if user has a group to invite users to
+var inviteGuard = {
+  guard (store) {
+    return !store.state.currentGroupId
+  },
+  redirect (to, from) {
+    return { path: '/new-group' }
+  }
+}
```

Comment:
An even shorter version that uses fewer braces and `return`s can be written like this:

```js
var loginGuard = {
  guard: store => !store.state.loggedIn,
  redirect: (to, from) => ({ path: '/signup', query: { next: to.path } })
}
// Check if user has a group to invite users to
var inviteGuard = {
  guard: store => !store.state.currentGroupId,
  redirect: (to, from) => ({ path: '/new-group' })
}
```